File: .NET/VS2010TrainingKit/Labs/WhatsNewInWCF4/Source/Ex4-ServiceDiscovery/End/C#/DiscoveryChat/PeerUser.cs
// ----------------------------------------------------------------------------------
// Microsoft Developer & Platform Evangelism
// 
// Copyright (c) Microsoft Corporation. All rights reserved.
// 
// THIS CODE AND INFORMATION ARE PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
// EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED WARRANTIES 
// OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
// ----------------------------------------------------------------------------------
// The example companies, organizations, products, domain names,
// e-mail addresses, logos, people, places, and events depicted
// herein are fictitious.  No association with any real company,
// organization, product, domain name, email address, logo, person,
// places, or events is intended or should be inferred.
// ----------------------------------------------------------------------------------

namespace Microsoft.Samples.Discovery
{
    using System.ServiceModel;

    public class PeerUser
    {
        private string userName;
        private EndpointAddress address;
        private ChatWindow chatWindow;

        public PeerUser(EndpointAddress address)
        {
            this.address = address;
        }

        public PeerUser(string userName, EndpointAddress address)
        {
            this.userName = userName;
            this.address = address;
        }

        public PeerUser(string userName, EndpointAddress address, ChatWindow chatWindow)
        {
            this.userName = userName;
            this.address = address;
            this.chatWindow = chatWindow;
        }

        public string UserName
        {
            get { return this.userName; }
        }

        public ChatWindow ChatWindow
        {
            get { return this.chatWindow; }
            set { this.chatWindow = value; }
        }

        public EndpointAddress Address
        {
            get { return this.address; }
            set { this.address = value; }
        }

        public bool ChatWindowExists
        {
            get { return (this.chatWindow != null) && !this.chatWindow.IsDisposed; }
        }

        public override string ToString()
        {
            if (this.userName == null)
            {
                return this.address.Uri.ToString();
            }
            else
            {
                return this.userName;
            }
        }
    }
}

File: .NET/VS2010TrainingKit/Labs/WhatsNewInWCF4/Source/Ex4-ServiceDiscovery/End/VB/DiscoveryChat/PeerUser.vb
' ----------------------------------------------------------------------------------
' Microsoft Developer & Platform Evangelism
' 
' Copyright (c) Microsoft Corporation. All rights reserved.
' 
' THIS CODE AND INFORMATION ARE PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
' EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED WARRANTIES 
' OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
' ----------------------------------------------------------------------------------
' The example companies, organizations, products, domain names,
' e-mail addresses, logos, people, places, and events depicted
' herein are fictitious.  No association with any real company,
' organization, product, domain name, email address, logo, person,
' places, or events is intended or should be inferred.
' ----------------------------------------------------------------------------------

Imports System.ServiceModel

Public Class PeerUser
    Private privateUserName As String

    Public Sub New(ByVal address As EndpointAddress)
        Me.Address = address
    End Sub

    Public Sub New(ByVal userName As String, ByVal address As EndpointAddress)
        Me.privateUserName = userName
        Me.Address = address
    End Sub

    Public Sub New(ByVal userName As String, ByVal address As EndpointAddress, ByVal chatWindow As ChatWindow)
        Me.privateUserName = userName
        Me.Address = address
        Me.ChatWindow = chatWindow
    End Sub

    Public ReadOnly Property UserName() As String
        Get
            Return Me.privateUserName
        End Get
    End Property

    Public Property ChatWindow() As ChatWindow

    Public Property Address() As EndpointAddress

    Public ReadOnly Property ChatWindowExists() As Boolean
        Get
            Return (Me.ChatWindow IsNot Nothing) AndAlso Not Me.ChatWindow.IsDisposed
        End Get
    End Property

    Public Overrides Function ToString() As String
        If Me.privateUserName Is Nothing Then
            Return Me.Address.Uri.ToString()
        Else
            Return Me.privateUserName
        End If
    End Function
End Class